app/routes/chatbot.js

```diff
@@ -112,74 +114,91 @@ router.post('/', (req, res) => {
       });
   });
 
-  const agentViewOip = process.env.MOBILECOMMONS_OIP_AGENTVIEW;
-
-  return loadUser
-    .then((user) => {
-      logger.info(`loaded user:${user._id}`);
-      scope.user = user;
-
-      let campaign;
-      let campaignId;
-
-      if (scope.broadcast_id) {
-        campaignId = process.env.CAMPAIGNBOT_BROADCAST_CAMPAIGN;
-        campaign = app.locals.campaigns[campaignId];
-
-        if (!campaign) {
-          const msg = `Broadcast Campaign '${campaignId}' not found.`;
-          throw new NotFoundError(msg);
-        }
-
-        // TODO: Add check on app start to trigger alert if Broadcast Campaign is closed.
-        if (campaign.status === 'closed') {
-          // TODO: Include this message to the CampaignClosedError.
-          const msg = `Broadcast Campaign ${campaignId} is closed.`;
-          throw new UnprocessibleEntityError(msg);
+  const loadCampaign = new Promise((resolve, reject) => {
+    logger.log('loadCampaign');
+    let campaign;
+    let campaignId;
+
+    return loadUser
+      .then((user) => {
+        logger.info(`loaded user:${user._id}`);
+        scope.user = user;
+
+        if (scope.broadcast_id) {
+          return app.locals.db.broadcasts
```

Comment:
Would it make sense to move this to a helper function, something like `loadBroadcastCampaign` to reduce nesting?